### PR TITLE
IMED-03: Edit Drug Screen Layout

### PR DESCRIPTION
**Pull request by Saotomefirst**
Changes:

- created Edit Drug screen
- extracted resources to dimens.xml and strings.xml

image

### CODE REVIEW

File: app/src/main/res/layout/fragment_drug_list.xml [hunk 1]
```diff
@@ -4,7 +4,8 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".ui.fragments.DrugListFragment">
+    tools:context=".ui.fragments.DrugListFragment"
+    >
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_drug_list.xml]: I think you can undo this change. I prefer ">" in the last line.

Linke this:

`tools:context=".ui.fragments.DrugListFragment">`

Maybe you should auto format the file.

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#66CCFF"
+    android:padding="16dp"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: Replace 16dp with a dimen resource.

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#66CCFF"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: Add `#66CCFF` to the colors.xml with a meaningful name.

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:text="Edit Drug" />
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/edit_drug_name_layout"
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: 16dp....

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            android:inputType="text"/>
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/edit_drug_dosage_layout"
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: 16dp....

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            android:inputType="text" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/edit_drug_start_time_layout"
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: 16dp....

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+            android:inputType="text" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/edit_drug_frequency_layout"
+        style="@style/Widget.MaterialComponents.TextInputLayout.OutlinedBox"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: 16dp....

File: app/src/main/res/layout/fragment_edit_drug.xml [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+            android:hint="Frequency"
+            android:inputType="text" />
+
+    </com.google.android.material.textfield.TextInputLayout>
+
+    <Button
+        android:id="@+id/edit_drug_save"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="8dp"
```
Comment (maylcf) [inline app/src/main/res/layout/fragment_edit_drug.xml]: 8dp....

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -9,4 +9,8 @@
 
     <string name="hello_first_fragment">Hello first fragment</string>
     <string name="hello_second_fragment">Hello second fragment. Arg: %1$s</string>
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (maylcf) [inline app/src/main/res/values/strings.xml]: Are you using `hello_first_fragment `or `hello_second_fragment `right now? If not, remove it.

If you are using it, something is not right with your pr.

File: .idea/misc.xml [hunk 10]
```diff
@@ -6,6 +6,9 @@
         <entry key="..\:/Users/jorge/StudioProjects/imed-app/app/src/main/res/layout/activity_main.xml" value="0.2" />
```
Comment (maylcf) [inline .idea/misc.xml]: I think you should add this file (.idea/misc.xml) to git ignore.

Please take a look at this link:
https://stackoverflow.com/questions/1274057/how-do-i-make-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore

File: app/src/main/java/com/saotome/imed_app/ui/fragments/EditDrug.kt [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.saotome.imed_app.R
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
```
Comment (maylcf) [inline app/src/main/java/com/saotome/imed_app/ui/fragments/EditDrug.kt:12]: Was all this code for handling arguments added when you created the fragment?

Comment (Saotomefirst) [reply app/src/main/java/com/saotome/imed_app/ui/fragments/EditDrug.kt:12]: Yes, this was all auto generated. I created the new fragment with its companion class just to have the structure ready. I haven't touched it yet - will do it on my next PR.

File: .idea/misc.xml (hunk 10 above)
Comment (Saotomefirst) [reply .idea/misc.xml]: Thanks for the tip! Done and dusted
40cc91830cda85344a87d640ad61f0a7b23ecedb

File: app/src/main/res/values/strings.xml (hunk 9 above)
Comment (Saotomefirst) [reply app/src/main/res/values/strings.xml]: Removed all references to unused  First and Second Fragments
61ec09ecb3e9ef5f2ff9e2add5e0b7e01deda10e

File: app/src/main/res/layout/fragment_drug_list.xml (hunk 1 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_drug_list.xml]: My bad. Done.
f29310a5ee674db86bbb64cf69872d5ade125327

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 2 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Dimen resource already exsited. Done.
81223eb832dd8f9d7ad51806b390a9f997067dd5

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 4 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Done.
81223eb832dd8f9d7ad51806b390a9f997067dd5

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 5 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Done.
81223eb832dd8f9d7ad51806b390a9f997067dd5

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 6 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Done.
81223eb832dd8f9d7ad51806b390a9f997067dd5

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 7 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Done.
81223eb832dd8f9d7ad51806b390a9f997067dd5

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 8 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Dimen resource already existes as well. Done.
81223eb832dd8f9d7ad51806b390a9f997067dd5

File: app/src/main/res/layout/fragment_edit_drug.xml (hunk 3 above)
Comment (Saotomefirst) [reply app/src/main/res/layout/fragment_edit_drug.xml]: Resource extracted.
81223eb832dd8f9d7ad51806b390a9f997067dd5

**Review by maylcf — APPROVED** [review]
👍

**Review by carlosheds — APPROVED** [review]
